pkg/reconciler/service/runner.go

```diff
@@ -80,22 +81,46 @@ func (r *runner) reconcile(ctx context.Context, model *reconciler.Reconciliation
 		return err
 	}
 
-	chartProvider, err := r.newChartProvider()
+	inClusterClient, err := kubeclient.NewInClusterClient()
+	if err != nil {
+		return err
+	}
+
+	inClusterClientSet, err := inClusterClient.GetClientSet()
+	if err != nil {
+		return err
+	}
+
+	clusterClientSet, err := kubeClient.Clientset()
+	if err != nil {
+		return err
+	}
+
+	configs := model.ConfigsToMap()
+	err = model.Repository.ReadToken(inClusterClientSet.CoreV1(), configs["repo.token.namespace"])
+	if err != nil {
+		return err
+	}
+
```

Comment:
Is this really globally required? Can't we move this part into the `connectiviy-proxy` itself?

Also the fields in the `ActionContext` are only required by the connectivity-proxy. Do we really have a need to make them available to all component reconcilers?

Reply:
To be handled on separate branch.

Reply:
Refactored.